doc/manuals/devel/httpHeaders.md

```diff
@@ -0,0 +1,353 @@
+9)  [X-Forwarded-For](#9-x-forwarded-for)                                                               
+10) [X-Real-IP](#10-x-real-ip)
+
+11) [Access-Control-Allow-Origin](#11--access-control-allow-origin)
+12) [Access-Control-Allow-Headers](#12-access-control-allow-headers)                                                                   
+13) [Access-Control-Allow-Methods](#13-access-control-allow-methods)
+14) [Access-Control-Max-Age](#14-access-control-max-age)                                 
+15) [Access-Control-Expose-Headers](#15-access-control-expose-headers)
+16) [Allow](#16-allow)                                                                                    
+17) [Location](#17-location)                                                 
+18) [Ngsiv2-AttrsFormat](#18-ngsiv2-attrsformat)                                    
+19) [Fiware-Service](#19-fiware-service)
+20) [Fiware-Servicepath](#20-fiware-servicepath)
+21) [Fiware-Total-Count](#21-fiware-total-count)
+22) [Fiware-Correlator](#22-fiware-correlator)                                     
+
+23) [X-Auth-Token](#23-x-auth-token)
+
+
+
+# HTTP Headers used by Orion with their Description:
+
+
+
+## 1. Accept
+
+Accept HTTP Header is used in requests. 
+Accept HTTP Header is used with the request URL for accepting the response in well defined JSON format. 
+For example in request URL if we use:
+	
+    curl ... 'Accept: application/json'
+		 
+On this request we'll get the response data in well defined JSON format.
+
+The API response payloads in this specification are based on `application/json` and  (for attribute value 
+type operation) `text/plain` MIME types. Clients issuing HTTP requests with accept types different 
+than those will get a `406 Not Acceptable` error.
+
+[Top](#top)
+
+## 2. Content-Length
+
+This header is used by both requests and responses.
+Content-length HTTP header is a mandatory header in Orion responses.
+it defines the length of the request and response body in octets (8-bit bytes).
+Orion Context Broker expects always a Content-Length header in all client requests, otherwise the client will receive a `411 Length Required` response. 
+This is due to the way the underlying HTTP library (microhttpd) works.
+	
+    Content-Length: 34	
+
+[Top](#top)
+ 
+## 3. Content-Type 
+
+It is used to send the requests in well defined JSON format.
+
+    curl ... 'Content-Type: application/json'
+
+This request will send data in well defined JSON format.
+
+Specifically, if the resulting URL after applying the template is malformed, then no notification is sent.
+
+[Top](#top)
+ 
+## 4. Connection
+ 
+ This Header is used by both requests and responses.
+ It is used as Control options for the current connection and list of hop-by-hop response fields.
+	
+    Connection: Keep-Alive
+    	
+[Top](#top)	
+  
+## 5. Expect
+
+Expect HTTP Header is used in requests.
+It Indicates that behaviour of particular server are required by the client.
```

Comment:
It seems the Expect header is only actually used at one point, in the httpRequestSendWithCurl() function: 

```
  // ----- Expect
  httpHeaderAdd(&headers, HTTP_EXPECT, " ", &outgoingMsgSize, extraHeaders, usedExtraHeaders);
```

Not fully sure, but I would say this causes a `Expect:` to be sent on the wire. Could you confirm?

Reply:
Yes, your understanding is correct. I confirmed the same as Expect HTTP header sends only `Expect:`.
In this [document](https://github.com/ArqumFarooqui-NEC/fiware-orion/blob/master/doc/manuals/devel/httpHeaders.md), the description of Expect HTTP header is provided in generalized form.

Reply:
I think the approach should be to include a link for the general behavior (e.g. https://tools.ietf.org/html/rfc7231#section-5.1.1) but explain here how Orion is using it in particular (i.e. the usage with blank value and explaining what that exactly implies for clients receiving the notification or forwarded request in which the header is included).

Reply:
FIxed in f76732e